src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -682,8 +771,7 @@ bool PgGate_GetDisableIndexBackfill() {
 }
 
 bool PgGate_IsK2PgEnabled() {
-//  return api_impl != nullptr;
-  return true;
+  return pg_gate != nullptr && pg_gate->IsInitialized();
```

Comment:
I thought this was going to come based on env variable. How else does the initialization code know to call InitPgGate() ?

Reply:
Yes, it is based on env variable. The logic is in k2pg_aux and it will check if the env variable is set, if it is set, it will initialize the pg gate, otherwise, it won't. For the PgGate_IsK2PgEnabled() calls, we only need to check using the above logic instead of checking the env variable every time.